build-packages.js

```diff
@@ -2,48 +2,73 @@ const fs = require("fs");
 const path = require("path");
 const del = require("del");
 const util = require("util");
+const { kebabCase, map } = require("lodash");
+const { peerDependencies } = require("./package.json");
 
 // Convert NodeJS methods to promises in order to use them with `await` statement
 const exec = util.promisify(require("child_process").exec);
 const writeFile = util.promisify(fs.writeFile);
+const makeDir = util.promisify(fs.mkdir);
 
 // Read available package sources from the dir
 const entryDirPath = path.join(__dirname, "src/packages");
 
 fs.readdir(entryDirPath, async (e, files) => {
-  console.log(`⚙️ Building ${files.length} packages...`);
-
-  for (let file of files) {
+  // Bundles a package asynchronously
+  const bundlePackage = async (file) => {
     const { name } = path.parse(file);
     const isMainPackage = name === "hex";
-    const filePath = path.join(entryDirPath, file);
     const outputDirPath = path.join(__dirname, isMainPackage ? "dist" : name);
     const manifestPath = path.join(outputDirPath, `package.json`);
     const bundlerPath = path.join(__dirname, "node_modules/.bin/microbundle");
 
+    // Format a package name according to NPM's naming guide
+    // https://docs.npmjs.com/files/package.json#name
+    const packageName = `react-colorful-${kebabCase(name)}`;
+
     // Delete the previous package version if exists
     await del(outputDirPath);
 
-    // Run microbundle
-    const { stdout } = await exec(
-      `${bundlerPath} --entry ${filePath} --output ${outputDirPath}/index.js  --name react-colorful-${name} --css-modules true --jsx React.createElement`
-    );
-    console.log(stdout);
-
     if (!isMainPackage) {
       // Create `package.json`
       var manifestCode = JSON.stringify({
-        name: `react-colorful-${name}`,
+        name: packageName,
         private: true,
         main: "index.js",
         module: "index.module.js",
         esmodule: "index.esmodule.js",
-        types: "../types/index.d.ts"
+        "umd:main": "dist/index.umd.js",
+        source: `../src/packages/${file}`,
+        types: "../types/index.d.ts",
+        peerDependencies,
       });
 
+      await makeDir(outputDirPath);
       await writeFile(manifestPath, manifestCode, "utf8");
     }
-  }
+
+    // Bundler options
+    const args = {
+      name: packageName,
+      cwd: isMainPackage ? __dirname : outputDirPath,
+      output: `${outputDirPath}/index.js`,
```

Comment:
```suggestion
      output: `${outputDirPath}`,
```

I don't think you need the `.js` here. [Microbundle's `output` is for a directory](https://github.com/developit/microbundle/#all-cli-options-)

Reply:
Whoops, didn't catch you in time. Sorry.

Reply:
Don't know how it works, but If I remove `index.js` microbundle names the files like that:
![image](https://user-images.githubusercontent.com/206567/91096874-6a708380-e667-11ea-8088-b8195836fce5.png)

Reply:
Looks like Microbundle doesn't look at `main: "index.js"` in `package.json`. Hmmm

Reply:
They use a file name in their own examples, so I think it's okay:
https://github.com/developit/microbundle/#building-a-single-bundle-with-a-fixed-output-name

Reply:
That's really odd. I don't think I've seen it doing that before. 

Though as I usually stick to TS, I guess I've avoided that issue. Nvm then. Don't know how that happens.